[GTK][WPE] Merge CompositingCoordinator into LayerTreeHost

carlosgcampos opened the pull request:
#### b0911c6ac3929da80947c6be57d70640428c6bd1
<pre>
[GTK][WPE] Merge CompositingCoordinator into LayerTreeHost
<a href="https://bugs.webkit.org/show_bug.cgi?id=280965">https://bugs.webkit.org/show_bug.cgi?id=280965</a>

Reviewed by Alejandro G. Castro.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp: Removed.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.h: Removed.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::setRootCompositingLayer):
(WebKit::LayerTreeHost::setViewOverlayRootLayer):
(WebKit::LayerTreeHost::forceRepaint):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::graphicsLayerFactory):
(WebKit::LayerTreeHost::didChangeViewport):
(WebKit::LayerTreeHost::detachLayer):
(WebKit::LayerTreeHost::attachLayer):
(WebKit::LayerTreeHost::paintingEngine):
(WebKit::LayerTreeHost::imageBackingStore):
(WebKit::LayerTreeHost::createGraphicsLayer):
(WebKit::LayerTreeHost::requestDisplayRefreshMonitorUpdate):
(WebKit::LayerTreeHost::requestUpdate):
(WebKit::LayerTreeHost::renderNextFrame):
(WebKit::LayerTreeHost::didFlushRootLayer): Deleted.
(WebKit::LayerTreeHost::updateScene): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
(WebKit::LayerTreeHost::LayerTreeHost): Deleted.
(WebKit::LayerTreeHost::~LayerTreeHost): Deleted.
(WebKit::LayerTreeHost::setLayerFlushSchedulingEnabled): Deleted.
(WebKit::LayerTreeHost::scheduleLayerFlush): Deleted.
(WebKit::LayerTreeHost::cancelPendingLayerFlush): Deleted.
(WebKit::LayerTreeHost::setRootCompositingLayer): Deleted.
(WebKit::LayerTreeHost::setViewOverlayRootLayer): Deleted.
(WebKit::LayerTreeHost::scrollNonCompositedContents): Deleted.
(WebKit::LayerTreeHost::forceRepaint): Deleted.
(WebKit::LayerTreeHost::forceRepaintAsync): Deleted.
(WebKit::LayerTreeHost::sizeDidChange): Deleted.
(WebKit::LayerTreeHost::pauseRendering): Deleted.
(WebKit::LayerTreeHost::resumeRendering): Deleted.
(WebKit::LayerTreeHost::graphicsLayerFactory): Deleted.
(WebKit::LayerTreeHost::contentsSizeChanged): Deleted.
(WebKit::LayerTreeHost::didChangeViewportAttributes): Deleted.
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged): Deleted.
(WebKit::LayerTreeHost::adjustTransientZoom): Deleted.
(WebKit::LayerTreeHost::commitTransientZoom): Deleted.
(WebKit::LayerTreeHost::createDisplayRefreshMonitor): Deleted.

Canonical link: <a href="https://commits.webkit.org/284817@main">https://commits.webkit.org/284817@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/839eb5da72567fed82c3a6db23bcc2e027e9d686

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/70441 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/49847 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23206 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/74530 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/21619 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57647 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/21459 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55821 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14291 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/73507 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45341 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60725 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36283 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42000 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19980 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63931 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18505 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76249 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/14667 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17742 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63550 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/14709 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60791 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63478 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11516 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5152 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10817 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/45649 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/416 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/46723 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48000 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/46465 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Great refactor!

carlosgcampos committed:
[GTK][WPE] Merge CompositingCoordinator into LayerTreeHost
https://bugs.webkit.org/show_bug.cgi?id=280965

Reviewed by Alejandro G. Castro.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp: Removed.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.h: Removed.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::setRootCompositingLayer):
(WebKit::LayerTreeHost::setViewOverlayRootLayer):
(WebKit::LayerTreeHost::forceRepaint):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::graphicsLayerFactory):
(WebKit::LayerTreeHost::didChangeViewport):
(WebKit::LayerTreeHost::detachLayer):
(WebKit::LayerTreeHost::attachLayer):
(WebKit::LayerTreeHost::paintingEngine):
(WebKit::LayerTreeHost::imageBackingStore):
(WebKit::LayerTreeHost::createGraphicsLayer):
(WebKit::LayerTreeHost::requestDisplayRefreshMonitorUpdate):
(WebKit::LayerTreeHost::requestUpdate):
(WebKit::LayerTreeHost::renderNextFrame):
(WebKit::LayerTreeHost::didFlushRootLayer): Deleted.
(WebKit::LayerTreeHost::updateScene): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
(WebKit::LayerTreeHost::LayerTreeHost): Deleted.
(WebKit::LayerTreeHost::~LayerTreeHost): Deleted.
(WebKit::LayerTreeHost::setLayerFlushSchedulingEnabled): Deleted.
(WebKit::LayerTreeHost::scheduleLayerFlush): Deleted.
(WebKit::LayerTreeHost::cancelPendingLayerFlush): Deleted.
(WebKit::LayerTreeHost::setRootCompositingLayer): Deleted.
(WebKit::LayerTreeHost::setViewOverlayRootLayer): Deleted.
(WebKit::LayerTreeHost::scrollNonCompositedContents): Deleted.
(WebKit::LayerTreeHost::forceRepaint): Deleted.
(WebKit::LayerTreeHost::forceRepaintAsync): Deleted.
(WebKit::LayerTreeHost::sizeDidChange): Deleted.
(WebKit::LayerTreeHost::pauseRendering): Deleted.
(WebKit::LayerTreeHost::resumeRendering): Deleted.
(WebKit::LayerTreeHost::graphicsLayerFactory): Deleted.
(WebKit::LayerTreeHost::contentsSizeChanged): Deleted.
(WebKit::LayerTreeHost::didChangeViewportAttributes): Deleted.
(WebKit::LayerTreeHost::deviceOrPageScaleFactorChanged): Deleted.
(WebKit::LayerTreeHost::adjustTransientZoom): Deleted.
(WebKit::LayerTreeHost::commitTransientZoom): Deleted.
(WebKit::LayerTreeHost::createDisplayRefreshMonitor): Deleted.

Canonical link: https://commits.webkit.org/284817@main
